rustfmt-core/rustfmt-lib/tests/target/issue-3988.rs

```diff
@@ -0,0 +1,10 @@
+// rustfmt-normalize_doc_attributes: true
+// rustfmt-max_width: 10
+
+mod a {
+    mod b {
+        ///test
+fn foo() -> (){
```

Comment:
This bug is caused by the case when the rustfmt fails to format because the width is greater than max_width in rustfmt.toml.
In this case, this line is greater than 10. rustfmt seems to remove the white space and emits the not formatted string. I think that's the reason why the implementation exists after comment on the same line.

Reply:
The indentation of `foo` looks/feels a bit odd :thinking: 

I feel like we may need a slightly different solution in order to maintain the target indent (since this exceeds the max_width anyway)